### DISCUSSION TITLE
Pre-compile subform fields

### DISCUSSION THREAD

**Discussion by PixedBo** [joomengine/Joomla-Component-Builder#875] [Q&A] [answered]
Hello, i would like to achieve this: the possiblity to pre-compile subform fields based on a dropdown choice. Like this:

![Cattura](https://user-images.githubusercontent.com/12549961/157068103-fd27ea7d-707b-41d8-b6a3-386c92be472d.JPG)

I have a view called "services" that has basically 2 fields:

- Name
- Cost per hour

I have created an admin view called "job" that has a subform 3 fields in it:

- Service (custom field that is filled from "services" entries)
- Hours (simple text input that asks how many hours the job did took)
- Total (this should take "cost per hour" from "services" view and multiply it for "hours" field of the subform)

I can handle the mulitply with a simple javascript, but i don't know how to retrieve in the subform the "cost per hour" of the selected service.

Thank you for your help!

**Comment by Llewellynvdm** [comment, marked as answer]
This is done with JavaScript and a good understanding of how the sub-form works... JCB has it all over the place so in terms of code you should be able to find examples.

Short answer... [subform has events](https://docs.joomla.org/Subform_form_field_type) and on those events you can act.

**Reply by PixedBo** [reply]
Ok thanks for the tip, i'll start from there
